django/contrib/auth/mixins.py

```diff
@@ -0,0 +1,85 @@
+from django.conf import settings
+from django.contrib.auth import REDIRECT_FIELD_NAME
+from django.contrib.auth.views import redirect_to_login
+from django.core.exceptions import ImproperlyConfigured, PermissionDenied
+from django.utils import six
+from django.utils.encoding import force_text
+
+
+class UserPassesTestMixin(object):
+    """
+    Mixin for class-based views that checks that the user passes the given by
+    the class method `test_func`.
+    """
+    login_url = None
+    raise_exception = False
```

Comment:
I had to go looking for the docs to figure out the purpose of this attribute. Better name? `raise_permission_denied_on_failure`